### PR TITLE
Bugfixes/string

### PR DESCRIPTION
**Pull request by SGSSGene**
This pull request fixes several issues when viewing `std::string`
1. It is possible to call `cppman string::string`
2. The link to `basic_string::npos` works now
3. formatting of  `erase/erase_if` inside of `std::string` is fixed
4. correcting formatting of `stoi, stol, stoll` inside of `std::string`
5. correct hightlighting of `std::swap` inside of `std::string`
6. updated data base for cppreference.com

Known issues - related to this PR:
- std::vector<bool> can't be viewed any more

probably unrelated issues:
- updating index database for cplusplus.com doesn't work (gets stuck on no_throw_t when crawling)

### CODE REVIEW

File: cppman/main.py [hunk 1]
```diff
@@ -131,15 +141,20 @@ def rebuild_index(self):
                 for title in results:
                     for (k, a) in results[title]["aliases"]:
                         sql_results = self.db_cursor.execute('SELECT id, keyword FROM "%s_keywords" '
-                                                             'WHERE keyword LIKE "%s::%%" OR keyword LIKE "%s" OR keyword LIKE "%s %%" OR keyword LIKE "%s)%%" OR keyword LIKE "%s,%%"' % (table, k, k, k, k, k)).fetchall()
+                                                             'WHERE keyword LIKE "%%::%s::%%" OR keyword LIKE "%s::%%" OR keyword LIKE "%s" OR keyword LIKE "%s %%" OR keyword LIKE "%s)%%" OR keyword LIKE "%s,%%"' % (table, k, k, k, k, k, k)).fetchall()
```
Comment (aitjcize) [inline cppman/main.py]: can we have a better line breaks here to keep the max col number to 80?

Comment (SGSSGene) [reply cppman/main.py]: Yes, of course.

**Comment by SGSSGene** [comment]
I am still a little unhappy with this pull request.
1. As in the description, it doesn't allow to open `std::vector<bool>`
2. I can't open the page for `std::fstream`

I can live with `std::vector<bool>` not working, but `std::fstream` is really anoying, I will look into it the next few days.

File: cppman/main.py [hunk 2]
```diff
@@ -113,33 +123,47 @@ def rebuild_index(self):
                                        '(id INTEGER NOT NULL PRIMARY KEY, title VARCHAR(255) NOT NULL UNIQUE, url VARCHAR(255) NOT NULL UNIQUE)' % table)
                 self.db_cursor.execute('CREATE TABLE "%s_keywords" '
                                        '(id INTEGER NOT NULL, keyword VARCHAR(255), FOREIGN KEY(id) REFERENCES "%s"(id))' % (table, table))
-                # crawl and insert all entries
```
Comment (aitjcize) [inline cppman/main.py]: can you also fix the above two queries as well?

**Comment by SGSSGene** [comment]
1. Now most lines in cppman/main.py should be fixed
2. cplusplus.com is fixed and can be crawled again
3. lookups for fstream work now

File: cppman/main.py (hunk 2 above)
Comment (SGSSGene) [reply cppman/main.py]: yes of course, I fixed a few more lines.

**Comment by aitjcize** [comment]
LGTM, sorry for the long delay, and thanks for fixing this :)

**Comment by aitjcize** [comment]
Can you also update the index database? We should release a new version soon.

**Pull request merged**